### PR TITLE
fix(dashmate): Dash Core container is unable to restart properly under WSL

### PR DESCRIPTION
**Pull request by jawid-h**
## Issue being fixed or feature implemented

Trying to solve WSL2 restart issue using dynamic ENV variable for Core container

## What was done?

- updated docker-compose file and generating ENV variable using `Date()`

## How Has This Been Tested?

local env

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/dashmate/src/docker/DockerCompose.js [hunk 1]
```diff
@@ -399,6 +399,7 @@ class DockerCompose {
       ...process.env,
       ...envs,
       DASHMATE_HOME_DIR: HOME_DIR_PATH,
+      CORE_WSL2_FIX: (new Date()).getTime(),
```
Comment (shumkov) [inline packages/dashmate/src/docker/DockerCompose.js:402]: We should do that only for WSL, not for other envs

File: packages/dashmate/docker-compose.yml [hunk 2]
```diff
@@ -13,6 +13,8 @@ services:
     command:
       - dashd
     stop_grace_period: 30s
+    environment:
+      - WSL2_FIX=${CORE_WSL2_FIX}
```
Comment (shumkov) [inline packages/dashmate/docker-compose.yml:17]: Please add comprehensive comment to describe the problem and the solution

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/dashmate/src/docker/DockerCompose.js (hunk 1 above)
Comment (shumkov) [reply packages/dashmate/src/docker/DockerCompose.js:404]: I guess this line needs to be removed

**Review by shumkov — CHANGES_REQUESTED** [review]

**Comment by jaboski** [comment]
Dashmate-helper crashing for me with:
`Cannot read properties of undefined (reading 'log')`
Sent you full log of the error. 
Steps:
- yarn dashmate setup local
- yarn dashmate group start

**Comment by pshenmic** [comment]
@jaboski sorry it was messed up by my last commits. this pr should resolve it 
https://github.com/dashpay/platform/pull/738

@jawid-h could you kindly pull this in?

**Comment by jawid-h** [comment]
@jaboski @pshenmic Merged #738

File: packages/dashmate/src/docker/DockerCompose.js [hunk 3]
```diff
@@ -401,6 +403,13 @@ class DockerCompose {
       DASHMATE_HOME_DIR: HOME_DIR_PATH,
     };
 
+    if (isWsl) {
+      // Solving issue under WSL when after restart container volume is not being mounted properly
+      // https://github.com/docker/for-win/issues/4812
+      // Following fix forces container recreation
+      env.CORE_WSL2_FIX = (new Date()).getTime();
```
Comment (pshenmic) [inline packages/dashmate/src/docker/DockerCompose.js:410]: I think `WSL2_FIX` is meant to be here as it will be applied to all containers

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — APPROVED** [review]
👍

**Comment by jaboski** [comment]
No more issue with crashing dashmate-helper. I also checked setup, start, stop, restart and reset for local - works fine for me 👍

**Pull request merged**